### ISSUE TITLE
JavaScript Errors on "keyup" to "Ensure any element in focus outside an overlay is visible"

### ISSUE THREAD

**Issue by RitchieD** [wet-boew/wet-boew#7091]
WET 4.0.17 using  IE 11

I’m getting WET JavaScript errors when submitting forms.  The problem occurs when the “keyup” event is triggered and no element has focus. 
An example scenario would be an AJAXed form.  If the user clicks enter while on an input field, the form is submitted and the AJAX content area is replaced with a wait gif. When the “keyup” eventually triggers no element has focus as that element has been cleared and the new content has no yet been displayed.
I’m seeing the error on non-AJAXed forms too, but can’t always reproduce or explain.

I propose a simple change to WET to avoid this issue.  See ****\* Proposed Change ****

Wet-boew.js

```
// Ensure any element in focus outside an overlay is visible
$document.on( "keyup", function( ) {
    var elmInFocus, elmInFocusRect, focusAreaBelow, focusAreaAbove,
    overlayId, overlay, overlayRect;

// Ignore if not initialized
if ( initialized ) {
    elmInFocus = document.activeElement;
    elmInFocusRect = elmInFocus.getBoundingClientRect();

    // ***** Proposed Change *****
    if (!elmInFocusRect)
        return;
    // ***** End of Proposed Change *****

    focusAreaBelow = 0;
    focusAreaAbove = window.innerHeight;

    // Ensure that at least one overlay is visible, and that the element in focus is not an overlay,
    // a child of an overlay, or the body element
    if ( $.isEmptyObject( sourceLinks ) || elmInFocus.className.indexOf( componentName ) !== -1 ||
        $( elmInFocus ).parents( selector ).length !== 0 || elmInFocus === document.body ) {
        return;
    }

    // Determine the vertical portion of the viewport that is not obscured by an overlay
    for ( overlayId in sourceLinks ) {
        overlay = document.getElementById( overlayId );
        if ( overlay && overlay.getAttribute( "aria-hidden" ) === "false" ) {
            overlayRect = overlay.getBoundingClientRect();
            if ( overlay.className.indexOf( "wb-bar-t" ) !== -1 ) {
                focusAreaBelow = Math.max( overlayRect.bottom, focusAreaBelow );
            } else if ( overlay.className.indexOf( "wb-bar-b" ) !== -1 ) {
                focusAreaAbove = Math.min( overlayRect.top, focusAreaAbove );
            }
        }
    }

    // Ensure the element in focus is visible
    // TODO: Find a solution for when there isn't enough page to scoll up or down
    if ( elmInFocusRect.top < focusAreaBelow ) {

        // Scroll down till the top of the element is visible
        window.scrollBy( 0, focusAreaBelow - elmInFocusRect.top );
    } else if ( elmInFocusRect.bottom > focusAreaAbove ) {

        // Scroll up till the bottom of the element is visible
        window.scrollBy( 0, elmInFocusRect.bottom - focusAreaAbove );
    }
}
} );    
```

**Comment by hsrudnicki** [comment]
@RitchieD @LaurentGoderre @masterbee Please update (help in) this issue's status whether this issue can be closed or otherwise?

**Comment by hsrudnicki** [comment]
Based on the above, this issue can be assumed as resolved but no any status update in regards to implementing the proposed change into WET, therefor again: Make "Going Stale" or otherwise?
Please update this issue' status whether this issue can be closed as fixed or otherwise?

**Comment by RitchieD** [comment]
As it stands now, I have to change WET code after downloading to avoid this error.  Why not modify the WET source to prevent the error?

**Comment by hsrudnicki** [comment]
Is this still an issue waiting for an answer?  Either way, please update this issue's status whether this can be closed as resolved/fixed or otherwise, thank you

**Comment by RitchieD** [comment]
As it stands now, I have to change WET code after downloading to avoid this error. Why not modify the WET source to prevent the error?

**Comment by hsrudnicki** [comment]
Is this still an issue? Please update the status of this for the current year, thank you

**Comment by RitchieD** [comment]
This is a suggested improvement to WET code to avoid client errors.  As a client of WET I should not change the WET code after I downloaded it.   I think a WET developer should address this issues as described above.    So no its not fixed.

**Comment by LaurentGoderre** [comment]
:+1:

**Comment by RitchieD** [comment]
This issue is fixed in WET 4.0.23.    Thank you.
